File: Examples/Cross Machine Platform/CS_WPF/OkumaInterface/API_Factory.cs
namespace CS_WPF.OkumaInterface
{
    public static class API_Factory
    {
        public static IOkuma GetOkumaAPI()
        {
            if(Okuma.Scout.Platform.BaseMachineType == Okuma.Scout.Enums.MachineType.L)
            {
                //If Lathe return new lathe wrapper
                return new OkumaLathe();
            }
            else if(Okuma.Scout.Platform.BaseMachineType == Okuma.Scout.Enums.MachineType.M)
            {
                //If machine is a mill return mill wrapper
                return new OkumaMill();
            }
            else
            {
                //If it is anything else return the simulator wrapper
                return new OkumaSim();
            }
        }
    }
}

File: Examples/Cross Machine Platform/VB_WPF/OkumaInterface/API_Factory.vb

Public Class API_Factory

    Public Shared Function GetAPI() As IOkuma

        If Okuma.Scout.Platform.BaseMachineType = Okuma.Scout.Enums.MachineType.L Then
            Return New OkumaLathe
        ElseIf Okuma.Scout.Platform.BaseMachineType = Okuma.Scout.Enums.MachineType.M Then
            Return New OkumaMill
        Else
            Return New OkumaSim
        End If

    End Function



End Class
